src/components/visualisations/BarChart.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { CartesianChartInput } from '../../types/ChartInput';
+import Plot from 'react-plotly.js';
+
+export type BarChartProps = {
+    barNames?: string[];
+    yLabel?: string;
+    title?: string;
+    data: CartesianChartInput;
+    isPreview?: boolean;
+    height: string | number;
+    width: string | number;
+};
+
+const BarChart: React.FC<BarChartProps> = ({ title, yLabel, data, isPreview = false, height, width }) => {
+    /*
+    data.map((el, index) => ({
+                type: 'bar',
+                x: el.map((d) => d.x),
+                y: el.map((d) => d.y),
+                name: barNames[index],
+            }))
+    */
```

Comment:
Don't commit comments.

If you wish to have them in the file, and commit the rest of the file I would recommend using chunked staging (available in most GUI Git clients – [example from Tower](https://www.git-tower.com/help/guides/faq-and-tips/tips-and-tricks/chunk-line-staging/windows)).